src/Education.UnitTests/RuleHelpXamlBuilderTests.cs

```diff
@@ -0,0 +1,98 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Education.XamlGenerator;
+using SonarLint.VisualStudio.Rules;
+
+namespace SonarLint.VisualStudio.Education.UnitTests
+{
+    [TestClass]
+    public class RuleHelpXamlBuilderTests
+    {
+        private static readonly Assembly ResourceAssembly = typeof(RuleMetadataProvider).Assembly;
+
+        [TestMethod]
+        public void CreateXamlString_CheckAllEmbedded()
```

Comment:
As discussed at the standup this morning, I have only written this one test.
It doesn't check the detail of the XAML that is produced.
However, it does attempt to build the XAML for all of the embedded rule help files, and checks that the builder does not fail for any of them.